single_tech_samples/datafactory/sample2_data_pre_processing_with_azure_batch/deploy/terraform/modules/azureBatch/main.tf

```diff
@@ -0,0 +1,156 @@
+resource "azurerm_batch_account" "batch_account" {
+  name                                = "${var.batch_account_name}${var.tags.environment}${var.batch_account_suffix}"
+  resource_group_name                 = var.resource_group_name
+  location                            = var.location
+  pool_allocation_mode                = var.pool_allocation_mode
+  storage_account_id                  = var.storage_account_id
+  storage_account_authentication_mode = var.storage_account_authentication_mode
+  storage_account_node_identity       = var.batch_uami_id
+  identity {
+    type = var.identity_type
+  }
+  tags = var.tags
+}
+
+resource "azurerm_public_ip" "orch_pool_ip" {
+  name                = "${var.orch_pool_name}-ip"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  allocation_method   = "Static"
+  sku                 = var.ip_sku
+  domain_name_label   = "${var.resource_group_name}-${azurerm_batch_account.batch_account.name}-${var.orch_pool_name}-ip"
```

Comment:
Can we rename it to something small? I used a slightly bigger resource group name and got the following error:

```
module.azure_batch.azurerm_batch_pool.exec_pool: Creation complete after 1m35s [id=/subscriptions/abc/resourceGroups/rg-databricks-sample/providers/Microsoft.Batch/batchAccounts/batchaccountdevmqxqqxej/pools/executionpool]
╷
│ Error: domain_name_label must contain only lowercase alphanumeric characters, numbers and hyphens. It must start with a letter, end only with a number or letter and not exceed 63 characters in length
│ 
│   with module.azure_batch.azurerm_public_ip.orch_pool_ip,
│   on modules/azureBatch/main.tf line 21, in resource "azurerm_public_ip" "orch_pool_ip":
│   21:   domain_name_label   = "${var.resource_group_name}-${azurerm_batch_account.batch_account.name}-${var.orch_pool_name}-ip"
```
```
"domain_name_label": "rg-databricks-sample-batchaccountdevmqxqqxej-executionpool-ip"
```